### ISSUE TITLE
[question] Tagging packages from beta versions of the compiler

### ISSUE THREAD

**Issue by NoWiseMan** [conan-io/conan#10660]
Currently we build most of our libraries using Conan on a Jenkins ci server. We build for all the platforms we support using different compilers and push them to Artifactory. Any changes we make to our Conan recipes are pushed to a test channel. When we merge those changes into the main channel, we rebuild the libraries and push them into the stable channel. So far, so good, everything is working well.
We are also trying to build with the latest beta compilers from Apple and Microsoft, so we need to add those compiler versions to settings.yml. 
We are currently deploying these builds in a beta channel. If we want to build our product with these beta compilers, we need to change our consuming conanfile.py to use this channel, which is cumbersome.
Alternatively, we could add an option to all conan recipes and add a tag to all BETA packages, or we could change settings.yml and add a 13.1beta compiler version, for example. What do you think is the best approach?

**Comment by lasote** [comment]
For these kinds of situations, we recommend using different Artifactory repositories and promoting packages between them when ready. You would need a (test, beta...) repositories, then the "user/channel" could be dropped. Could that help? You could use also lock files to add extra guarantees of the packages being used.

**Comment by NoWiseMan** [comment]
Different Artifactory repositories within 1 server?

`You could use also lock files to add extra guarantees of the packages being used.`
I need to look into this .. thx for your feedback

**Comment by memsharded** [comment]
> Different Artifactory repositories within 1 server?

Yes, this is a recommended good practice in general. For example in CI (https://docs.conan.io/en/latest/versioning/lockfiles/ci.html#repositories), you don't want to put packages that are being built in the main repo, because they might break something. So they are put in another "build" Artifactory repo, to reuse them during the build and testing. And only once everything is looking good, they are copied to the main "develop" Artifactory repo.

One of recommended practices in devops is "immutability", and we have agreed to go this way for conan 2.0, where it will not be possible to "conan copy" packages from one channel to another. It is important that once one artifact has been built and tested, it is not rebuilt (because something could go wrong in the new build). So the recommended practice by devops experts in the server "promotion" mechanism in which packages are copied from one repository to another depending on the stage.

**Comment by NoWiseMan** [comment]
Ok, good to know that you will go this way for 2.0. Thanks a lot, even if we have to make a lot of changes in our approach for that.

**Comment by AbrilRBS** [comment]
Closing as solved with Conan 2
